packages/core/src/store/mad-config/hooks.ts

```diff
@@ -1,9 +1,26 @@
 import { getPremadeSettings } from "../../utils/getPremadeSettings";
 import { useMadConfig } from "./mad-config";
+import { getRedirectUriFromAuthConfig } from "../../utils/getRedirectUriFromAuthConfig";
+import { MadConfig } from "../../types";
 
-export const useAuthConfig = () => {
-    const config = useMadConfig();
-    return config.authentication;
+export type AuthConfig = {
+    clientId: MadConfig["authentication"]["clientId"]
+    scopes?: MadConfig["authentication"]["scopes"]
+    /**
+     * resolved redirectUri. This can either be `redirectUri` or `redirectUriWeb` from the mad config, depending on which platform
+     * you're running on.
+     */
+    redirectUri: string
+}
```

Comment:
Would it be better to set `AuthConfig` to the following to have the optional props in sync?

```TS
export type AuthConfig = Pick<MadConfig["authentication"], "clientId" | "scopes"> & {redirectUri: string}
```

Reply:
No that will be wrong. scopes should not be optional. We will always return scopes. Fixed